components/tidb_query_vec_executors/src/index_scan_executor.rs

```diff
@@ -232,59 +321,133 @@ impl IndexScanExecutorImpl {
+                    .mut_decoded()
+                    .push_int(Some(handle));
+            }
+            DecodeCommonHandle => {
+                key = &key[table::PREFIX_LEN + table::ID_LEN..];
+                datum::skip_n(&mut key, self.columns_id_without_handle.len())?;
+                Self::extract_columns_from_datum_format(
+                    &mut key,
+                    &mut columns[self.columns_id_without_handle.len()..self.schema.len()],
+                )?;
+            }
         }
         Ok(())
     }
 
-    fn process_kv_pair_old(
+    fn process_normal_old_collation_value(
         &mut self,
         key: &[u8],
         value: &[u8],
         columns: &mut LazyBatchColumnVec,
     ) -> Result<()> {
-        check_index_key(key)?;
+        use DecodeHandleStrategy::*;
+
         // The payload part of the key
         let mut key_payload = &key[table::PREFIX_LEN + table::ID_LEN..];
 
-        for i in 0..self.columns_id_without_handle.len() {
-            let (val, remaining) = datum::split_datum(key_payload, false)?;
-            columns[i].mut_raw().push(val);
-            key_payload = remaining;
-        }
+        Self::extract_columns_from_datum_format(
+            &mut key_payload,
+            &mut columns[0..self.columns_id_without_handle.len()],
+        )?;
 
-        if self.decode_handle {
+        match self.decode_strategy {
+            NoDecode => {}
             // For normal index, it is placed at the end and any columns prior to it are
             // ensured to be interested. For unique index, it is placed in the value.
-            let handle_val = if key_payload.is_empty() {
+            DecodeIntHandle if key_payload.is_empty() => {
                 // This is a unique index, and we should look up PK handle in value.
-
-                self.decode_handle_from_value(value)?
-            } else {
-                // This is a normal index. The remaining payload part is the PK handle.
-                // Let's decode it and put in the column.
-
-                self.decode_handle_from_key(key_payload)?
-            };
-
-            columns[self.columns_id_without_handle.len()]
-                .mut_decoded()
-                .push_int(Some(handle_val));
+                let handle_val = self.decode_handle_from_value(value)?;
+                columns[self.columns_id_without_handle.len()]
+                    .mut_decoded()
+                    .push_int(Some(handle_val));
+            }
+            DecodeIntHandle => {
+                let handle_val = self.decode_handle_from_key(key_payload)?;
+                columns[self.columns_id_without_handle.len()]
+                    .mut_decoded()
+                    .push_int(Some(handle_val));
+            }
+            DecodeCommonHandle => {
+                Self::extract_columns_from_datum_format(
+                    &mut key_payload,
+                    &mut columns[self.columns_id_without_handle.len()..self.schema.len()],
```

Comment:
The pattern `self.columns_id_without_handle.len()..self.schema.len()` is frequently occured, how about assigning a variable for it?

Reply:
We could implement the trait `Index<RangeFrom<usize>>` for `LazyColumnVec`